### DISCUSSION TITLE
automation tokens cannot create dist-tags

### DISCUSSION THREAD

**Discussion by gr2m** [npm/feedback#108] [Registry] [answered]
We run into yet another problem with the new [automation tokens](https://github.blog/changelog/2020-10-02-npm-automation-tokens/) when used with [semantic-release](https://semantic-release.gitbook.io/semantic-release/). Enabling the `whoami` enabled us to support the automation tokens, thanks for that 🙏🏼 But semantic-release also supports release workflows with [pre-release, next, and maintenance channels](https://semantic-release.gitbook.io/semantic-release/usage/workflow-configuration), all for which dist-tags are used. If the dist-tags such as `release-15.x`, `next`, or `beta` don't yet exist, they need to be created. Unfortunately, the automation tokens does not permit to created dist channels at this point as confirmed by @lapa182 (https://github.com/semantic-release/npm/issues/298#issuecomment-737185327)

I think managing dist-tags is within the scope of automation tokens, as it's part of release automation to support maintenance and preview releases. Do you think you could enable the dist-tags endpoint to be usable with automation tokens, at least the creation of the dist-tags?

**Comment by MylesBorins** [comment]
Hey @gr2m

I believe that dist-tags should indeed be in scope. I pinged @ethomson and we'll dig into this

**Reply by lapa182** [reply]
Hey @MylesBorins is there anywhere where we can follow the progress or help out somehow? The team was asking me for an update 😅.

**Reply by MylesBorins** [reply]
Hey @lapa182 

Thanks for the ping. We've added this to the public roadmap and will aim to get it out next quarter. We will keep [the issue](https://github.com/npm/roadmap/issues/29) up to date

**Comment by MylesBorins** [comment, marked as answer]
We've added an item to the [public roadmap](https://github.com/npm/roadmap/issues/29) and have this work on track to be done next quarter.

**Reply by MylesBorins** [reply]
As an update a fix has been implemented and pushed to production, please try it out!

**Comment by thislooksfun** [comment]
@MylesBorins it seems that automation tokens can't update the 'latest' tag either. Running the same command (`npm dist-tag add snoots@1.0.0-dev.6 <tag>`) with an automation token succeeds when `<tag>` is `dev`, but fails (403) if `<tag>` is `latest`.

**Reply by MylesBorins** [reply]
Thanks for reporting. I've let the engineering team know about this

**Reply by MylesBorins** [reply]
@thislooksfun we pushed a fix to production, can you take a look and lmk if it works for you now

**Reply by thislooksfun** [reply]
@MylesBorins Looks like it works! `npm dist-tag add snoots@1.0.0-dev.X latest` runs perfectly with an automation token.

**Comment by estherkim** [comment]
Hi @MylesBorins! I'm running into the same 403 issue using an automation token. But what's interesting is that it works for some packages but not others. For example:
```
npm dist-tag add @ampproject/amp-timeago@1.2106030132.0 latest // works!
npm dist-tag add @ampproject/amp-youtube@1.2106030132.0 latest // does not work!
```
The publishing access for both packages are the same. Any ideas? Thanks :)

**Reply by MylesBorins** [reply]
Thanks for reporting @estherkim

I've opened an issue with engineering

**Reply by estherkim** [reply]
Confirmed that this works now. Thank you!
